src/ArduinoCellular.h

```diff
@@ -119,9 +119,15 @@ class ArduinoCellular {
          * @param apn The Access Point Name.
          * @param username The APN username.
          * @param password The APN password.
+         * @param waitForever The function does not return unless a connection has been established
          * @return True if the connection is successful, false otherwise.
          */
-        bool connect(String apn = "", String username = "", String password = "");
+        bool connect(String apn = "", String username = "", String password = "", bool waitForever = true);
+        
+        /**
+         * @brief same as previous, username and password are empty
+         */
+        bool connect(String apn, bool waitForever);
```

Comment:
```suggestion
        bool connect(String apn, bool waitForever = true);
```
I'd assign the same default value to the version without credentials.